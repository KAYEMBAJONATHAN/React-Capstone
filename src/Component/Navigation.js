import React from 'react';
import { Link, Outlet } from 'react-router-dom';
import back from '../Assets/back.png';
import mic from '../Assets/mic.png';
import setting from '../Assets/setting.png';
import '../Style/navigation.css';

const Navigation = () => (
  <>
    <div className="country">
      <Link to="/"><img src={back} alt="back" /></Link>
      <h1>Countries</h1>
      <div>
        <img src={mic} alt="mic" />
        <img src={setting} alt="setting" />
      </div>
    </div>
    <Outlet />
  </>
);

export default Navigation;
