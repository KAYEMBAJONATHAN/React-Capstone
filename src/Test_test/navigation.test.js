import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from '../Component/Navigation';

describe('Navigation', () => {
  it('Should renders correctly', () => {
    const component = renderer.create(
      <Router>
        <Navigation />
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
