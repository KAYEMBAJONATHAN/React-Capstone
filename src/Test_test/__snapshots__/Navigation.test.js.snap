// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Should renders correctly 1`] = `
<div
  className="country"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="back"
      src="back.png"
    />
  </a>
  <h1>
    Countries
  </h1>
  <div>
    <img
      alt="mic"
      src="mic.png"
    />
    <img
      alt="setting"
      src="setting.png"
    />
  </div>
</div>
`;
